bolts/data/datasets/audio/base.py

```diff
@@ -1,11 +1,67 @@
 from __future__ import annotations
 from pathlib import Path
 
+from kit import implements
+import numpy as np
+import numpy.typing as npt
+from torch import Tensor
+import torchaudio
+
 from bolts.data.datasets.base import PBDataset
+from bolts.data.datasets.utils import (
+    AudioLoadingBackend,
+    AudioTform,
+    apply_waveform_transform,
+    infer_al_backend,
+)
+from bolts.data.structures import TargetData
+
+__all__ = ["PBAudioDataset"]
 
 
 class PBAudioDataset(PBDataset):
-    """Base Dataset for audio data."""
+    """Base dataset for audio data."""
+
+    x: npt.NDArray[np.string_]
+
+    def __init__(
+        self,
+        *,
+        x: npt.NDArray[np.string_],
+        audio_dir: Path | str,
+        y: TargetData | None = None,
+        s: TargetData | None = None,
+        transform: AudioTform | None = None,
+    ) -> None:
+        super().__init__(x=x, y=y, s=s)
+
+        # Convert string path to Path object.
+        if isinstance(audio_dir, str):
+            audio_dir = Path(audio_dir)
+
+        self.audio_dir = audio_dir
+        self.transform = transform
+
+        # Infer the appropriate audio-loading backend based on the operating system.
+        self.al_backend: AudioLoadingBackend = infer_al_backend()
+        self.log(f'Using {self.al_backend} as backend for audio-loading')
+        torchaudio.set_audio_backend(self.al_backend)
+
+    def __repr__(self) -> str:
+        head = "Dataset " + self.__class__.__name__
+        body = [f"Number of datapoints: {len(self)}"]
+        body.append(f"Base audio-directory location: {self.audio_dir.resolve()}")
+        body += self.extra_repr().splitlines()
+        if hasattr(self, "transform") and self.transform is not None:
+            body += [repr(self.transform)]
+        lines = [head] + [" " * self._repr_indent + line for line in body]
+        return '\n'.join(lines)
+
+    def load_waveform(self, index: int) -> Tensor:
+        return torchaudio.load(self.audio_dir / self.x[index])
 
-    def __init__(self) -> None:
-        pass
+    @implements(PBDataset)
+    def _sample_x(self, index: int) -> Tensor:
```

Comment:
Even though x is always a tensor, the method's signature needs to include coerce_to_tensor as this is set to true when computing the shape of x in the superclass.

Reply:
Ahh ok, yea I had just assumed since it's always a tensor there's no need.  Fixed, will check on things like this in future.